lightly/data/_video.py

```diff
@@ -388,4 +388,26 @@ def get_filename(self, index):
         else:
             n_frames = self.__len__() - self.offsets[i]
         
-        return f'{video_name}-{frame_number:0{len(str(n_frames))}}-{video_format}.png'
\ No newline at end of file
+        return f'{video_name}-{frame_number:0{len(str(n_frames))}}-{video_format}.png'
+
+    def get_filenames(self) -> List[str]:
+        """Returns a list filenames for all frames in the dataset.
+        
+        """
+        filenames = []
+        for i, filename in enumerate(self.videos):
+            filename = os.path.relpath(filename, self.root)
+            splits = filename.split('.')
+            video_format = splits[-1]
+            video_name = '.'.join(splits[:-1])
+
+            if i < len(self.offsets) - 1:
+                n_frames = self.offsets[i+1] - self.offsets[i]
+            else:
+                n_frames = len(self) - self.offsets[i]
+
+            for frame_number in range(n_frames):
+                filenames.append(
+                    f'{video_name}-{frame_number:0{len(str(n_frames))}}-{video_format}.png'
```

Comment:
I guess you could precompute `len(str(n_frames))` as well

Reply:
for that extra nanosecond saved ;)

Reply:
Added it :)